src/core.ts

```diff
@@ -6,7 +6,7 @@ const MultiServer = require('multiserver')
 const Inactive = require('pull-inactivity')
 const debug = require('debug')('secret-stack')
 
-function isPlainObject (o: any): any {
+function isPlainObject (o: unknown): o is Record<string, unknown> {
```

Comment:
Returning `o is Record<string, unknown>` is almost the same thing as returning a boolean, but it's called a "type guard" and is a hint we give to TypeScript that after this function is applied, it can **known** that `o` is at least a `Record<string, unknown>`, allowing us to do `console.log(o.x)` without compilation errors (whereas `console.log(o.x)` would originally be a compilation error because `o` is `unknown`).

Essentially, we're matching "runtime error possibility" with "compilation error possibility". Since this function checks **in runtime** that `o` is an object that can have any of its properties inspected (`o.x` or `o.y` or `o.z` or whatever), we also want TypeScript to know in compile-time that we can access its properties.

Reply:
PS: a `Record<string, unknown>` is essentially any object that has `string` properties (not e.g. numbers, not symbols) and `unknown` values. It's a key-value object, where keys are strings and values are unknown.

Reply:
nice, almost haskell